src/inference/src/os/win/win_system_conf.cpp

```diff
@@ -77,65 +84,73 @@ void parse_processor_info_win(const char* base_ptr,
-                _proc_type_table[0] = line_value_0;
+                _proc_type_table[0] = proc_init_line;
             }
         } else if (info->Relationship == RelationProcessorCore) {
             MaskToList(info->Processor.GroupMask->Mask);
 
-            if (proc_count >= _processors) {
-                break;
-            }
-
             if (0 == list[0]) {
-                base_proc = proc_count;
+                base_proc = _processors;
             }
 
             if (2 == list_len) {
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_PROCESSOR_ID] = list[0] + base_proc;
-                _cpu_mapping_table[list[1] + base_proc][CPU_MAP_PROCESSOR_ID] = list[1] + base_proc;
-
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_SOCKET_ID] = _sockets;
-                _cpu_mapping_table[list[1] + base_proc][CPU_MAP_SOCKET_ID] = _sockets;
-
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_CORE_ID] = _cores;
-                _cpu_mapping_table[list[1] + base_proc][CPU_MAP_CORE_ID] = _cores;
-
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_CORE_TYPE] = HYPER_THREADING_PROC;
-                _cpu_mapping_table[list[1] + base_proc][CPU_MAP_CORE_TYPE] = MAIN_CORE_PROC;
-
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_GROUP_ID] = group;
-                _cpu_mapping_table[list[1] + base_proc][CPU_MAP_GROUP_ID] = group;
+                proc_info = cpu_init_line;
+                proc_info[CPU_MAP_PROCESSOR_ID] = list[0] + base_proc;
+                proc_info[CPU_MAP_SOCKET_ID] = _sockets;
+                proc_info[CPU_MAP_CORE_ID] = _cores;
+                proc_info[CPU_MAP_CORE_TYPE] = HYPER_THREADING_PROC;
+                proc_info[CPU_MAP_GROUP_ID] = group;
+                _cpu_mapping_table.push_back(proc_info);
+
+                proc_info = cpu_init_line;
+                proc_info[CPU_MAP_PROCESSOR_ID] = list[1] + base_proc;
+                proc_info[CPU_MAP_SOCKET_ID] = _sockets;
+                proc_info[CPU_MAP_CORE_ID] = _cores;
+                proc_info[CPU_MAP_CORE_TYPE] = MAIN_CORE_PROC;
+                proc_info[CPU_MAP_GROUP_ID] = group;
+                _cpu_mapping_table.push_back(proc_info);
 
                 _proc_type_table[0][MAIN_CORE_PROC]++;
                 _proc_type_table[0][HYPER_THREADING_PROC]++;
                 group++;
 
             } else {
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_PROCESSOR_ID] = list[0] + base_proc;
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_SOCKET_ID] = _sockets;
-                _cpu_mapping_table[list[0] + base_proc][CPU_MAP_CORE_ID] = _cores;
+                proc_info = cpu_init_line;
+                proc_info[CPU_MAP_PROCESSOR_ID] = list[0] + base_proc;
+                proc_info[CPU_MAP_SOCKET_ID] = _sockets;
+                proc_info[CPU_MAP_CORE_ID] = _cores;
+                if ((_processors > group_start) && (_processors <= group_end)) {
```

Comment:
Could you confirm _processors belongs to (group_start, group_end] or [group_start, group_end)?

Reply:
(group_start, group_end]. Processor of group_start has been handled before create group_start and group_end.